Tests/Pcap++Test/Tests/LiveDeviceTests.cpp

```diff
@@ -97,54 +96,106 @@ class RpcapdServerInitializer
 {
 private:
 	HANDLE m_ProcessHandle;
+	HANDLE m_JobHandle;
 
-public:
+	void killProcessAndCloseHandles()
+	{
+		if (m_ProcessHandle != nullptr)
+		{
+			TerminateProcess(m_ProcessHandle, 0);
+			CloseHandle(m_ProcessHandle);
+			m_ProcessHandle = nullptr;
+		}
 
-	RpcapdServerInitializer(bool activateRemoteDevice, const std::string &ip, uint16_t port) : m_ProcessHandle(nullptr)
+		if (m_JobHandle != nullptr)
+		{
+			CloseHandle(m_JobHandle);
+			m_JobHandle = nullptr;
+		}
+	}
+public:
+	RpcapdServerInitializer(bool activateRemoteDevice, const std::string& ip, uint16_t port)
+		: m_ProcessHandle(nullptr), m_JobHandle(nullptr)
 	{
 		if (!activateRemoteDevice)
 			return;
 
 		std::string cmd = "rpcapd\\rpcapd.exe";
-		std::ostringstream args;
-		args << "rpcapd\\rpcapd.exe -b " << ip << " -p " << port << " -n";
+		std::string args;
+		// Reserves the size of the arguments string in 1 allocation.
+		// Sizeof C-string includes the NULL terminator in the size.
+		// For the purposes of this calculation the NULL terminator's inclusion in the size is used to simulate the space delimiter after the argument.
+		args.reserve(
+			sizeof("rpcapd\\rpcapd.exe -b") + ip.size() +
+			sizeof(" -p") + 5 /* The maximum digits a uint16_t can have is 5 */ +
+			sizeof(" -n") - 1 /* Subtracts one as the last NULL terminator is not needed */
+		);
+		args += "rpcapd\\rpcapd.exe -b ";
+		args += ip;
+		args += " -p ";
+		args += std::to_string(port);
+		args += " -n";
+
+		m_JobHandle = CreateJobObject(nullptr, nullptr);
+		if (m_JobHandle == nullptr)
+		{
+			throw std::runtime_error("Failed to create a job object with error code: " + std::to_string(GetLastError()));
+		}
+
+		// Sets up the job limits so closing the job will automatically kill all processes assigned to the job.
+		// This will prevent the subprocess continuing to live if the current process is killed without unwinding the stack (i.e. std::terminate is called),
+		// as the OS itself will kill the subprocesses when the last job handle is closed.
+		JOBOBJECT_EXTENDED_LIMIT_INFORMATION jobLimitInfo;
```

Comment:
I am sure the following Unix works, but not sure for Windows:
`JOBOBJECT_EXTENDED_LIMIT_INFORMATION jobLimitInfo = {}`?

Reply:
Implemented 334a4d0